### PR TITLE
Added SubList exercise

### PR DESCRIPTION
**Pull request by ryanplusplus**
See the [description](https://github.com/exercism/x-common/blob/master/sublist.md)

### CODE REVIEW

File: exercises/sublist/example.lua [hunk 1]
```diff
@@ -0,0 +1,12 @@
+return function(sublist, list)
+  if #sublist == 0 then return true end
+
+  for i = 1, #list - #sublist + 1 do
+    for j = 1, #sublist do
+      if list[i + j - 1] ~= sublist[j] then break end
+      if j == #sublist then return true end
```
Comment (aarti) [inline exercises/sublist/example.lua]: small optimization by switching line 6 and 7 and return early, otherwise looks great.

Comment (ryanplusplus) [inline exercises/sublist/example.lua]: Actually that will cause failures by not checking for the last element to be a match.

Comment (aarti) [inline exercises/sublist/example.lua]: Ya, you are right.

**Comment by aarti** [comment]
:+1:

**Comment by ryanplusplus** [comment]
Thanks @aarti  :)

**Pull request merged**